internal/app_test.go

```diff
@@ -1,228 +1,181 @@
+	if !testUsingHardware() {
+		t.Skipf("Hardware not configured for tests")
+	}
+
+	require := require.New(t)
 
-func Test_FindLedger(t *testing.T) {
 	app, err := FindLedgerOasisApp()
-	if err != nil {
-		t.Fatalf(err.Error())
-	}
+	require.NoError(err, "FindLedgerOasisApp")
+	require.NotNil(app, "Must find a ledger device and initialize the interface")
 
-	assert.NotNil(t, app)
 	defer app.Close()
 }
 
-func Test_UserGetVersion(t *testing.T) {
-	app, err := FindLedgerOasisApp()
-	if err != nil {
-		t.Fatalf(err.Error())
-	}
+func TestUserGetVersion(t *testing.T) {
+	require, assert := require.New(t), assert.New(t)
+
+	app, err := testFindLedgerOasisApp()
+	require.NoError(err, "FindLedgerOasisApp")
 	defer app.Close()
 
 	version, err := app.GetVersion()
-	require.Nil(t, err, "Detected error")
-	fmt.Println(version)
+	require.NoError(err, "GetVersion")
+
+	t.Logf("Version: %s", version)
 
-	assert.Equal(t, uint8(0x0), version.AppMode, "TESTING MODE ENABLED!!")
-	assert.Equal(t, uint8(0x0), version.Major, "Wrong Major version")
-	assert.Equal(t, uint8(0xd), version.Minor, "Wrong Minor version")
+	assert.Equal(uint8(0x0), version.AppMode, "TESTING MODE ENABLED!!")
+	assert.Equal(uint8(0x0), version.Major, "Wrong Major version")
+	assert.Equal(uint8(0xd), version.Minor, "Wrong Minor version")
 }
 
-func Test_UserGetPublicKey(t *testing.T) {
-	app, err := FindLedgerOasisApp()
-	if err != nil {
-		t.Fatalf(err.Error())
-	}
+func TestUserGetPublicKey(t *testing.T) {
+	require := require.New(t)
+
+	app, err := testFindLedgerOasisApp()
+	require.NoError(err, "FindLedgerOasisApp")
 	defer app.Close()
 
 	path := []uint32{44, 474, 5, 0, 21}
 
 	pubKey, err := app.GetPublicKeyEd25519(path)
-	if err != nil {
-		t.Fatalf("Detected error, err: %s\n", err.Error())
-	}
+	require.NoError(err, "GetPublicKeyEd25519")
 
-	assert.Equal(t, 32, len(pubKey),
-		"Public key has wrong length: %x, expected length: %x\n", pubKey, 32)
-	fmt.Printf("PUBLIC KEY: %x\n", pubKey)
+	t.Logf("Public key: %x\n", pubKey)
 
-	assert.Equal(t,
-		"d71c79ffd5a6d438de89c833e00222a2e80ed94e9929350ef7c1c97d1d13295d",
-		hex.EncodeToString(pubKey),
-		"Unexpected pubkey")
+	checkTestKey(t, pubKey, "", path)
 }
 
-func Test_GetAddressPubKeyEd25519_Zero(t *testing.T) {
-	app, err := FindLedgerOasisApp()
-	if err != nil {
-		t.Fatalf(err.Error())
-	}
+func TestGetAddressPubKeyEd25519_Zero(t *testing.T) {
+	require := require.New(t)
+
+	app, err := testFindLedgerOasisApp()
+	require.NoError(err, "FindLedgerOasisApp")
 	defer app.Close()
 
 	path := []uint32{44, 474, 0, 0, 0}
 
 	pubKey, addr, err := app.GetAddressPubKeyEd25519(path)
-	if err != nil {
-		t.Fatalf("Detected error, err: %s\n", err.Error())
-	}
-
-	fmt.Printf("PUBLIC KEY : %x\n", pubKey)
-	fmt.Printf("BECH32 ADDR: %s\n", addr)
+	require.NoError(err, "GetAddressPublicKeyEd25519")
 
-	assert.Equal(t, 32, len(pubKey), "Public key has wrong length: %x, expected length: %x\n", pubKey, 32)
+	t.Logf("Public key: %x\n", pubKey)
```

Comment:
Maybe these log statements should also be moved into the `checkTestKey` helper since they repeat in mostly all invocations of the helper?

Reply:
Fixed.